plugins/nodes/src/js/data/MesosMasters.js

```diff
@@ -1,84 +1,83 @@
 import React from "react";
-import { Observable } from "rxjs/Observable";
-import "rxjs/add/observable/timer";
-import "rxjs/add/operator/do";
-import "rxjs/add/operator/map";
-import "rxjs/add/operator/retry";
-import "rxjs/add/operator/filter";
 
-import MesosStateStore from "#SRC/js/stores/MesosStateStore";
-import { findNestedPropertyInObject } from "#SRC/js/utils/Util";
+import { mesosMastersLeader } from "./MesosMastersLeader";
+import { mesosMastersHealth } from "./MesosMastersHealth";
 
 import LeaderGrid from "../components/LeaderGrid";
+import NonLeadersGrid from "../components/NonLeadersGrid";
 
-function isEmptyObject(obj) {
-  return Object.keys(obj).length === 0;
+function mastersInitialState() {
+  return {
+    leader: {
+      hostPort: undefined,
+      hostIp: undefined,
+      version: undefined,
+      electedTime: undefined,
+      startTime: undefined,
+      region: undefined
+    },
+    masters: undefined
+  };
 }
 
-function hostPort(address) {
-  return `${address.hostname}:${address.port}`;
-}
+function filterLeader(leader, masters) {
+  if (masters === undefined) {
+    return undefined;
+  }
 
-export function getRegion(master) {
-  return (
-    findNestedPropertyInObject(master, "domain.fault_domain.region.name") ||
-    "N/A"
-  );
+  return masters.filter(master => master.host_ip !== leader.hostIp);
 }
 
-const STORE_POLL_INTERVAL = 2000;
+export function combineMasterData(leaderDataSource, mesosMastersHealth) {
+  const mesosLeader$ = leaderDataSource().startWith(
+    mastersInitialState().leader
+  );
+  const mesosMasters$ = mesosMastersHealth().startWith(undefined);
 
-function mesosMasterEmpty() {
-  return {
-    hostPort: undefined,
-    version: undefined,
-    electedTime: undefined,
-    startTime: undefined,
-    region: undefined
+  return function() {
+    return mesosLeader$.combineLatest(mesosMasters$, (leader, masters) => ({
+      leader,
+      masters: filterLeader(leader, masters)
+    }));
   };
 }
 
-export function mesosMasterInfo(
-  masterDataSource,
-  interval = STORE_POLL_INTERVAL
-) {
-  return Observable.timer(0, interval)
-    .map(_ => {
-      return masterDataSource();
-    })
-    .map(mesosState => mesosState.master_info)
-    .filter(master => !isEmptyObject(master))
-    .map(master => {
-      return {
-        hostPort: hostPort(master.address),
-        version: master.version,
-        electedTime: master.elected_time,
-        startTime: master.start_time,
-        region: getRegion(master)
-      };
-    });
-}
-
 // This is an attempt of the mediator pattern without componentFromStream
-export default class MesosMasters extends React.Component {
-  constructor() {
-    super(...arguments);
+export function connectMasterComponent(initialState, stream) {
+  class MesosMasters extends React.Component {
```

Comment:
Why not use componentFromStream?

Reply:
I wanna try something different more inline with @weblancaster ideas of connect. I though I could expend 10 minutes trying that.